src/common/procedure/src/local.rs

```diff
@@ -412,14 +457,33 @@ impl ProcedureManager for LocalManager {
             }
         }
 
+        if !finished_ids.is_empty() {
+            logging::info!(
+                "LocalManager try to clean finished procedures, num: {}",
+                finished_ids.len()
+            );
+
+            for procedure_id in finished_ids {
+                if let Err(e) = procedure_store.delete_procedure(procedure_id).await {
+                    logging::error!(e; "Failed to delete procedure {}", procedure_id);
+                }
+            }
+        }
+
+        logging::info!("LocalManager finish recovery");
+
         Ok(())
     }
 
     async fn procedure_state(&self, procedure_id: ProcedureId) -> Result<Option<ProcedureState>> {
+        self.manager_ctx.remove_outdated_meta(META_TTL);
+
         Ok(self.manager_ctx.state(procedure_id))
     }
 
     fn procedure_watcher(&self, procedure_id: ProcedureId) -> Option<Watcher> {
+        self.manager_ctx.remove_outdated_meta(META_TTL);
```

Comment:
Why not use a `RepeatedTask` to purge outdated states periodically? Then we don't have to add so many hooks in code.

Reply:
I just don't want to implement the TaskFunction trait. I'll fix it.

Reply:
Using `RepeatedTask` requires adding `start()` and `stop()` to the procedure manager.

Reply:
I plan to leave this as a good first issue. I realize that the `RepeatedTask` doesn't need an async start function. I'd refactor it in another PR.